### ISSUE TITLE
Main search field not returning results for Third parties or Contacts

### ISSUE THREAD

**Issue by wrightjs** [Dolibarr/dolibarr#16125]
# Instructions
*This is a template to help you report good issues. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *replace the bracket enclosed texts with meaningful information*
- *remove any unused sub-section*

# Bug
[Main search field - top left of screen not returning results for Third parties or Contacts]

## Environment
- **Version**: [13.0.0]
- **OS**: [x86_64 Linux]
- **Web server**: [Apache 2.4.46]
- **PHP**: [7.2.34]
- **Database**: [MySQL 5.7.32]

## Expected and actual behavior
[Using the main search field - top left of interface and expecting to get results returned. No results returned, blank screen]

## Steps to reproduce the behavior
[Enter characters of a third party that I know is stored in the database and press Return or click on Third parties on drop down]

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
[*Files*
![image](https://user-images.githubusercontent.com/78321296/106396924-cef7f180-6402-11eb-8863-b94ec13b2ea8.png)
]

**Comment by atm-maxime** [comment]
Blank screen or normal screen with an empty thirdparty list ? If blank, check your Apache error log file. If empty, check if other filters are present (status open, ...)

**Comment by wrightjs** [comment]
Hi, thank you.

Blank screen and nothing in the Apache error log.

I've done some more testing..

I was searching "peacock" when the blank screen was returned.

Searching for "p" returns results

Searching "pe" returns results

Searching "pea" returns blank screen.

Kind regards,

**Comment by eldy** [comment]
Can you try to increase memory limit of php page.
Check again error.log. Blank screen should return a line into error.log file.

**Comment by wrightjs** [comment]
Thank you Laurent,

I have increased the memory limit. Issue still present.

Further use has revealed the same blank screen issue after creating a new project task.

I have attached log files for the two modules.

Cordialement,
Jerry

[societe error_log.txt](https://github.com/Dolibarr/dolibarr/files/5912122/societe.error_log.txt)
[projet error_log.txt](https://github.com/Dolibarr/dolibarr/files/5912124/projet.error_log.txt)

**Comment by ksar-ksar** [comment]
Hi,

You have error : PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 68359824 bytes) in /mydomain.co.uk/core/lib/images.lib.php on line 521
Need to increase the PHP memory

Also you have warnings PHP Warning:  Cannot modify header information - headers already sent by (output started at /mydomain.co.uk/core/tpl/extrafields_list_array_fields.tpl.php:3) in /mydomain.co.uk/main.inc.php on line 1202
Need to put dolibarr_prod = 1 in conf.php

**Comment by wrightjs** [comment]
Hi, thank you.

The PHP Fatal error line is old and I don't believe is relevant to this error. I have increased PHP memory up to 1G but still have the issue.

The reference to the conf.php file, please confirm syntax: is it this $dolibarr_main_prod='1';

The PHP error that is generated with the blank screen is:-

[09-Feb-2021 19:03:19 UTC] PHP Warning:  Cannot modify header information - headers already sent by (output started at /mydomain.co.uk/core/tpl/extrafields_list_array_fields.tpl.php:3) in /mydomain.co.uk/societe/list.php on line 535

**Comment by ksar-ksar** [comment]
> The reference to the conf.php file, please confirm syntax: is it this $dolibarr_main_prod='1';

Should be like this : https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/conf/conf.php.example#L241

**Comment by wrightjs** [comment]
Great, thank you. Line now added to conf.php

**Comment by caos30** [comment]
@eldy i've the same problem (blank web browser page after do a search on the left column form), and i found the problem. At least in my case is another one that the reported above, but very more simple to fix. Also it happens only sometimes (you will understand it):

the problem is that the file `/htdocs/core/tpl/extrafields_list_array_fields.tpl.php` has a PRINT command which is immediately "echoed" to browser, generating the typical error: "**PHP Warning:  Cannot modify header information - headers already sent by...**"

```
<?php

//print '<!-- extrafields_list_array_fields.tpl.php -->'."\n";

// Protection to avoid direct call of template
if (empty($conf) || !is_object($conf))
{
	print "Error, template page can't be called as URL";
	exit;
}

```
The solution is to comment or delete that line... fully unnecessary ;-)
Eldy, i have mentioned to you because i'm almost sure that you can comment that line directly on that core file because this bug is so simple that it's better not to open new request issue, it isn't?

Upsss... i see that the same happens with other files on that directory!
For example this other: `extrafields_list_search_input.tpl.php`
So, maybe you will be able to better fix all them and alert developer who introduced this bug for avoid it in the future.

Note: i'm working with Dolibarr v.13.0.1

**Comment by TheTrueTom** [comment]
> The solution is to comment or delete that line... fully unnecessary ;-)

I can confirm that this solves the blank page bug after a search ! 
Tested on Dolibarr v13.0.0

**Comment by leninrivas** [comment]
Thankyou. Solved Dolibarr 13.0.0
